diff --git a/src/config/proxygroup.h b/src/config/proxygroup.h
--- a/src/config/proxygroup.h
+++ b/src/config/proxygroup.h
@@ -32,11 +32,13 @@ struct ProxyGroupConfig
     Integer Interval = 0;
     Integer Timeout = 0;
     Integer Tolerance = 0;
+    Integer MaxFailedTimes = 0;
     BalanceStrategy Strategy = BalanceStrategy::ConsistentHashing;
     Boolean Lazy;
     Boolean DisableUdp;
     Boolean Persistent;
     Boolean EvaluateBeforeUse;
+    String ExpectedStatus;
 
     String TypeStr() const
     {
diff --git a/src/config/binding.h b/src/config/binding.h
--- a/src/config/binding.h
+++ b/src/config/binding.h
@@ -57,10 +57,14 @@ namespace toml
                 conf.Type = ProxyGroupType::Fallback;
                 conf.Url = find<String>(v, "url");
                 conf.Interval = find<Integer>(v, "interval");
+                conf.Tolerance = find_or<Integer>(v, "tolerance", 0);
+                if(v.contains("lazy"))
+                    conf.Lazy = find_or<bool>(v, "lazy", false);
                 if(v.contains("evaluate-before-use"))
                     conf.EvaluateBeforeUse = find_or(v, "evaluate-before-use", conf.EvaluateBeforeUse.get());
                 break;
@@ -79,7 +83,15 @@ namespace toml
                 throw serialization_error(format_error("Proxy Group has unsupported type!", v.at("type").location(), "should be one of following: select, url-test, load-balance, fallback, relay, ssid"), v.at("type").location());
             }
-            conf.Timeout = find_or(v, "timeout", 5);
+            conf.Timeout = find_or(v, "timeout", 0);
+            conf.MaxFailedTimes = find_or<Integer>(v, "max-failed-times", 0);
+            if(v.contains("expected-status"))
+            {
+                if(v.at("expected-status").is_integer())
+                    conf.ExpectedStatus = std::to_string(find<Integer>(v, "expected-status"));
+                else
+                    conf.ExpectedStatus = find<String>(v, "expected-status");
+            }
             conf.Proxies = find_or<StrArray>(v, "rule", {});
             conf.UsingProvider = find_or<StrArray>(v, "use", {});
             if(conf.Proxies.empty() && conf.UsingProvider.empty())
diff --git a/src/generator/config/subexport.cpp b/src/generator/config/subexport.cpp
--- a/src/generator/config/subexport.cpp
+++ b/src/generator/config/subexport.cpp
@@ -627,15 +627,25 @@ void proxyToClash(std::vector<Proxy> &nodes, YAML::Node &yamlnode, const ProxyGr
             [[fallthrough]];
         case ProxyGroupType::URLTest:
-            if(!x.Lazy.is_undef())
-                singlegroup["lazy"] = x.Lazy.get();
             [[fallthrough]];
         case ProxyGroupType::Fallback:
+            if(!x.Lazy.is_undef())
+                singlegroup["lazy"] = x.Lazy.get();
             singlegroup["url"] = x.Url;
             if(x.Interval > 0)
                 singlegroup["interval"] = x.Interval;
             if(x.Tolerance > 0)
                 singlegroup["tolerance"] = x.Tolerance;
+            if(x.Timeout > 0)
+                singlegroup["timeout"] = x.Timeout;
+            if(x.MaxFailedTimes > 0)
+                singlegroup["max-failed-times"] = x.MaxFailedTimes;
+            if(!x.ExpectedStatus.empty())
+                singlegroup["expected-status"] = x.ExpectedStatus;
+            if(!x.EvaluateBeforeUse.is_undef())
+                singlegroup["evaluate-before-use"] = x.EvaluateBeforeUse.get();
             break;
         default:
             continue;
